File: src/main/java/br/com/jiratorio/service/IssueService.java
package br.com.jiratorio.service;

import br.com.jiratorio.domain.estimate.EstimateIssue;
import br.com.jiratorio.domain.sandbox.SandBox;
import br.com.jiratorio.domain.sandbox.SandBoxFilter;
import br.com.jiratorio.domain.entity.Board;
import br.com.jiratorio.domain.entity.Issue;
import br.com.jiratorio.domain.form.IssueForm;
import java.util.List;

public interface IssueService {

    List<Issue> createByJql(String jql, Board board);

    List<EstimateIssue> findEstimateByJql(String jql, Board board);

    SandBox findByExample(Long boardId, IssueForm issueForm);

    List<Long> findLeadTimeByExample(Long boardId, IssueForm issueForm);

    SandBoxFilter findSandBoxFilters(Long boardId, SandBox sandBox, IssueForm issueForm);

    List<Issue> findByIssuePeriodId(Long issuePeriodId);

    void deleteAll(List<Issue> issues);
}

File: src/main/kotlin/br/com/jiratorio/service/IssueService.kt
package br.com.jiratorio.service

import br.com.jiratorio.domain.entity.Board
import br.com.jiratorio.domain.entity.Issue
import br.com.jiratorio.domain.request.SearchIssueRequest
import br.com.jiratorio.domain.response.ListIssueResponse
import br.com.jiratorio.domain.response.issue.IssueDetailResponse
import br.com.jiratorio.domain.response.issue.IssueKeysResponse
import br.com.jiratorio.domain.response.issue.IssueFilterResponse
import java.time.LocalDate

interface IssueService {

    fun findByExample(
        boardId: Long,
        dynamicFilters: Map<String, Array<String>>,
        searchIssueRequest: SearchIssueRequest
    ): ListIssueResponse

    fun findLeadTimeByExample(board: Board, searchIssueRequest: SearchIssueRequest): List<Long>

    fun deleteAll(issues: Set<Issue>)

    fun findByBoardAndId(boardId: Long, id: Long): IssueDetailResponse

    fun findFilters(boardId: Long): IssueFilterResponse

    fun findFilterKeys(boardId: Long, startDate: LocalDate, endDate: LocalDate): IssueKeysResponse

}
